pkg/index/index.go

```diff
@@ -222,6 +237,69 @@ func (m *MergedIndexes) IndexByEntry(entry *Entry) *Index {
 	return m.indexByEntry[entry]
 }
 
+// ParseReference is a helper function that parse with the followig logic:
+//
+//  1. if name is the name of an artifact, it will use the merged index to compute
+//     its reference. The tag latest is always appended.
+//     e.g "cloudtrail" -> "ghcr.io/falcosecurity/plugins/cloudtrail:latest"
+//     if instead a tag or a digest is specified, the name will be used to look up
+//     into mergedIndexes, then the tag or digest will be appended.
+//     e.g "cloudtrail:0.5.1" -> "ghcr.io/falcosecurity/plugins/cloudtrail:0.5.1"
+//     e.g "cloudtrail@sha256:123abc..." -> "ghcr.io/falcosecurity/plugins/cloudtrail@sha256:123abc...
+//
+//  2. if name is a reference without tag or digest, tag latest is appended.
+//     e.g. "ghcr.io/falcosecurity/plugins/cloudtrail" -> "ghcr.io/falcosecurity/plugins/cloudtrail:latest"
+//
+//  3. if name is a complete reference, it will be returned as is.
+func (m *MergedIndexes) ParseReference(name string) (string, error) {
```

Comment:
```suggestion
func (m *MergedIndexes) ResolveReference(name string) (string, error) {
```

This function is probably misnamed. It seems that it takes a relative name or a partial reference and resolves it to a fully qualified artifact reference. Thus my renaming suggestion.

Reply:
Makes sense!